src/amo/reducers/addonsByAuthors.js

```diff
@@ -1,125 +1,139 @@
 /* @flow */
-import type { AddonType, ExternalAddonType } from 'core/types/addons';
+import invariant from 'invariant';
+
 import { createInternalAddon } from 'core/reducers/addons';
+import type { AddonType, ExternalAddonType } from 'core/types/addons';
 
 
 type State = {
   byAddonSlug: { [string]: AddonType },
+  byUserId: { [number]: Array<AddonType> },
 };
 
 export const initialState: State = {
   byAddonSlug: {},
+  byUserId: {},
 };
 
-export const OTHER_ADDONS_BY_AUTHORS_PAGE_SIZE = 6;
+export const ADDONS_BY_AUTHORS_PAGE_SIZE = 6;
 
 // For further information about this notation, see:
 // https://github.com/mozilla/addons-frontend/pull/3027#discussion_r137661289
-export const FETCH_OTHER_ADDONS_BY_AUTHORS: 'FETCH_OTHER_ADDONS_BY_AUTHORS'
-  = 'FETCH_OTHER_ADDONS_BY_AUTHORS';
-export const LOAD_OTHER_ADDONS_BY_AUTHORS: 'LOAD_OTHER_ADDONS_BY_AUTHORS'
-  = 'LOAD_OTHER_ADDONS_BY_AUTHORS';
+export const FETCH_ADDONS_BY_AUTHORS: 'FETCH_ADDONS_BY_AUTHORS'
+  = 'FETCH_ADDONS_BY_AUTHORS';
+export const LOAD_ADDONS_BY_AUTHORS: 'LOAD_ADDONS_BY_AUTHORS'
+  = 'LOAD_ADDONS_BY_AUTHORS';
 
-type FetchOtherAddonsByAuthorsParams = {|
+type FetchAddonsByAuthorsParams = {|
   addonType: string,
   authors: Array<string>,
   errorHandlerId: string,
-  slug: string,
+  excludeAddonBySlug?: string,
```

Comment:
The original function was used was like this: "show me add-ons by other authors to display on the add-on I'm currently viewing." 

Thus, it's confusing to call this `excludeAddonBySlug` because the logic of excluding this slug from the results is just an implementation detail. I think calling it `slug` is still fine but maybe `addonSlug` would be more explicit. You could also call it `forAddonSlug` to make it more clear that you want to fetch a bunch of other add-ons for the sole purpose of this one add-on slug that the user is viewing.

It kinda makes sense to think of this as an exclusion parameter when *fetching* the add-ons but it no longer makes sense when loading the add-ons or doing any other reducer action. It still makes sense to group them in state like `addonsByAuthors.byAddonSlug` because that's saying here are some other add-ons from the perspective of add-on slug `X`.

Reply:
Yeah, I was thinking in terms of `fetch`ing but agree that it's awkward elsewhere. `forAddonSlug` is totally the right name and fits everything, merci !